bin/faucet/src/faucet/mod.rs

```diff
@@ -445,24 +462,37 @@ fn parse_desync_error(err: &str) -> Result<Digest, anyhow::Error> {
         .map(Into::into)
 }
 
+/// A collection of `P2Id` notes.
 struct P2IdNotes(Vec<Note>);
 
 impl P2IdNotes {
-    fn build(source: FaucetId, requests: &[MintRequest], rng: &mut impl FeltRng) -> Self {
-        Self(requests
-            .iter()
-            .filter_map(|request| {
-                // SAFETY: source is definitely a faucet account, and the amount is valid.
-                let asset = FungibleAsset::new(source.inner(), request.asset_amount.inner()).unwrap();
-                create_p2id_note(
-                    source.inner(),
-                    request.account_id,
-                    vec![asset.into()],
-                    request.note_type.into(),
-                    Felt::default(),
-                    rng,
-                ).inspect_err(|err| tracing::error!(request.account_id=%request.account_id, ?err, "failed to build note")).ok()
-            }).collect())
+    /// Builds a collection of `P2Id` notes from a set of mint requests.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if creating any p2id note fails.
+    fn build(
+        source: FaucetId,
+        requests: &[MintRequest],
+        rng: &mut impl FeltRng,
+    ) -> Result<Self, MintError> {
+        // TODO: if building a note fails, we discard the whole batch. Is this error possible?
```

Comment:
I think it's fine to assume `create_p2id_note` should not fail here, but I don't mind if you prefer keeping the error either